src/services/RTC.ts

```diff
@@ -284,6 +327,19 @@ class RTC {
         }
         return false;
     };
+
+    handleReceiveData = (event: MessageEvent) => {
+        this.receiveBuffer.push(event.data);
+        this.receivedSize += event.data.byteLength;
+
+        if (this.receivedSize >= this.receivingFileMaxSize) {
```

Comment:
btw, i know this could cause bugs if server ends up being slower than the RTC connection; I'll polish it up in next iteration.

Reply:
What do you mean? Isn't this using data just from the RTC connection and no socket interaction?

Reply:
`this.receivingFileMaxSize` is a value that's sent over the socket connection before it is set in the `RTC` class. So if we get enough of the file over RTC before we know the size of the file, we will get bugs.

Reply:
I see now, maybe we can hold off on sending the data until the receiver receives the file information? Just thinking out loud.

Reply:
Ya that's what i was gonna add in the next one